### PR TITLE
Revisit run tests

### PR DESCRIPTION
**Pull request by muryanto1**
Adding Lib/TestRunnerBase.py Lib/Util.py and Lib/Const.py.

### CODE REVIEW

File: Lib/TestRunnerBase.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        os.chdir("uvcdat-testdata")
+        ret_code, cmd_output = self.__run_cmd("git pull")
+        if ret_code != SUCCESS:
+            return ret_code
+        print("BRANCH WE ARE TRYING TO CHECKOUT is (%s)" % branch)
+        ret_code, cmd_output = self.__run_cmd("git checkout %s" % (branch))
+        os.chdir(workdir)
+        return(ret_code)
+
+    def __install_vtk_cdat(self):
```
Comment (doutriaux1) [inline Lib/TestRunnerBase.py:167]: I wonder if we should tweak this to let it install any package name from a specific chanel and version?

File: Lib/TestRunnerBase.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        else:
+            self.ncpus = 1
+
+        if args.attributes:
+            self.nosetests_attr = args.attributes
+        else:
+            self.nosetests_attr = []
+
+        if get_sample_data == True:
+            download_sample_data_files(os.path.join(sys.prefix, "share",
```
Comment (doutriaux1) [inline Lib/TestRunnerBase.py]: let's have the file name sent as part of base class initialization

File: Lib/TestRunnerBase.py [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+            opts += ["-A", att]
+            command = ["nosetests", ] + opts + ["-s", test_name]
+            start = time.time()
+            ret_code, out = run_command(command, True, verbosity)
+            end = time.time()
+            return {test_name: {"result": ret_code, "log": out, "times": {
+                "start": start, "end": end}}}
+
+    def run_nose_NEW(self, arguments):
+        run_options, attrs, verbosity, test_name = arguments
```
Comment (doutriaux1) [inline Lib/TestRunnerBase.py]: should we pass this as arguments for simplicity?

File: Lib/TestRunnerBase.py (hunk 2 above)
Comment (muryanto1) [reply Lib/TestRunnerBase.py]: Done. 
Made the  test_data_files_info to be an argument of the TestRunnerBase's __init()

File: Lib/TestRunnerBase.py (hunk 1 above)
Comment (muryanto1) [reply Lib/TestRunnerBase.py:167]: I removed this function as I am not sure if I need it.

**Pull request merged**